sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuShuffleExchangeExecBase.scala

```diff
@@ -81,6 +81,13 @@ class GpuShuffleMeta(
 
   override def tagPlanForGpu(): Unit = {
 
+    val supportedShuffleOrigins: Seq[ShuffleOrigin] = Seq(ENSURE_REQUIREMENTS, REPARTITION_BY_COL,
+      REPARTITION_BY_NUM, REBALANCE_PARTITIONS_BY_NONE, REBALANCE_PARTITIONS_BY_COL)
```

Comment:
This should not be here, as the sequence will be reconstructed on every instance yet has no variance per instance.  This should be moved to `object GpuShuffleMeta` and arguably should be a hash set rather than a sequence.